### ISSUE TITLE
APT error: cannot find repository

### ISSUE THREAD

**Issue by robinpaulson** [Shnatsel/cjdns-ubuntu-pubkey#4]
Unable to connect to ubuntu.repo.cjdns.ca:http:
W: Failed to fetch http://ubuntu.repo.cjdns.ca/dists/trusty/Release.gpg  Unable to connect to ubuntu.repo.cjdns.ca:http:

etc.

**Comment by Shnatsel** [comment]
The company that has providing the server is no longer in a position where they can provide them for free, and I'm not in a position to actually rent a server. So the repo is down. Permanently, unless someone steps up to either sponsor a VPS or host a repo themselves. The latter is preferred because I no longer actively use cjdns. 

Also, I'll open-source the build scripts I've written for cjdns if anyone's interested.

**Comment by robinpaulson** [comment]
On 2 December 2016 12:03:56 NZDT, Shnatsel <notifications@github.com> 
wrote:
> The company that has providing the server is no longer in a position
> where they can provide them for free, and I'm not in a position to
> actually rent a server. So the repo is down. Permanently, unless
> someone steps up to either sponsor a VPS or host a repo themselves. The
> latter is preferred because I no longer actively use cjdns.
> 
> Also, I'll open-source the build scripts I've written for cjdns if
> anyone's interested.

I may be able to host the repo, it depends on requirements. Typically, 
how much disk/internet data is required?

--
robin

**Comment by Shnatsel** [comment]
At about a gigabyte of disk storage for pbuilder chroots. It does need root privileges for chroot to work, although it might be possible to work around that.

CPU requirements are really low because it's essentially a daily cronjob plus a webserver to serve the repo (which is a bunch of static files). Build process won't use lots of RAM either.

I'm not sure about the bandwidth requirements, but last time I checked it barely used my 100Mbit connection, so it should not be a hog... I guess.

**Comment by Shnatsel** [comment]
I've published the scripts I've written for the repo to https://github.com/Shnatsel/cjdns-daily-build

It has occurred to me that we can get away with just the cronjob to preprocess source code and get rid of the web server and pbuilder chroots entirely if we use Launchpad's PPA system. We still can't get daily builds to work entirely though Launchpad, but we might get it to perform the builds and to host the resulting files.

**Comment by robinpaulson** [comment]
I'm in favour of moving to Launchpad. Also, do daily builds matter? I don't recall seeing a new build every day for a while now....

**Comment by wfleurant** [comment]
If we can get funds to the admins can we maintain what was in place?

**Comment by Shnatsel** [comment]
Yes. But before that I'll try hooking into Launchpad from my personal machine. That's gonna get us more transparency and result in a simpler, more secure setup. We can move the cronjob that's uploading things to Launchpad to a VPS later if need be.

**Comment by Shnatsel** [comment]
Progress report: I have a prototype script that prepares the source code and uploads it to a repository on Launchpad. You can check out the (experimental) result at https://launchpad.net/~shnatsel/+archive/ubuntu/staging, so far for 16.04 only

**Comment by Shnatsel** [comment]
The repository on a custom server should be now superseded by this PPA: https://launchpad.net/~cjdns-ppa/+archive/ubuntu/ppa

To anyone interested in seeing this through: please test that the packages from that PPA and report whether they work for you. Once they do, I'll update the instructions in the README.

**Comment by vladns** [comment]
During the installation package of the new repository, an error occurs.
`root@ubuntu:~# apt-get install cjdns
Чтение списков пакетов… Готово
Построение дерева зависимостей
Чтение информации о состоянии… Готово
Предлагаемые пакеты:
  cjdns-dynamic
НОВЫЕ пакеты, которые будут установлены:
  cjdns
обновлено 0, установлено 1 новых пакетов, для удаления отмечено 0 пакетов, и 0 пакетов не обновлено.
Необходимо скачать 309 kБ архивов.
После данной операции, объём занятого дискового пространства возрастёт на 818 kB.
Пол:1 http://ppa.launchpad.net/cjdns-ppa/ppa/ubuntu xenial/main amd64 cjdns amd64 0.18.0+r4303+xenial1~git6781edd [309 kB]
Получено 309 kБ за 0с (3 590 kБ/c)
Выбор ранее не выбранного пакета cjdns.
(Чтение базы данных … на данный момент установлено 94672 файла и каталога.)
Подготовка к распаковке …/cjdns_0.18.0+r4303+xenial1~git6781edd_amd64.deb …
Распаковывается cjdns (0.18.0+r4303+xenial1~git6781edd) …
Обрабатываются триггеры для ureadahead (0.100.0-19) …
Настраивается пакет cjdns (0.18.0+r4303+xenial1~git6781edd) …
/usr/bin/deb-systemd-helper: error: unable to read cjdns-resume.service
Failed to get unit file state for cjdns-resume.service: No such file or directory
cjdns-resume.service is a disabled or a static unit, not starting it.
cjdns.service is a disabled or a static unit, not starting it.
Обрабатываются триггеры для ureadahead (0.100.0-19) …
root@ubuntu:~#
`

**Comment by wfleurant** [comment]
yes i remember this causing issues elsewhere re: `cjdns-resume.service`

**Comment by wfleurant** [comment]
also, what address should resolve for `ubuntu.shnatsel.repo.cjdns.ca`

**Comment by vladns** [comment]
About the error during installation:
https://github.com/cjdelisle/cjdns/pull/1045/commits/ec087203c0922ab5b6bf92838d5e3c881385e75e
